subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/TransformParameters.java

```diff
@@ -21,13 +21,30 @@
 /**
  * Marker interface for parameter objects to {@link TransformAction}s.
  *
+ * <p>
+ *     Parameter types should use be interfaces, only declaring getters for {@link org.gradle.api.provider.Property} like objects.
+ *     All getters need to be annotated one of the input annotations {@link org.gradle.api.tasks.Console}, {@link org.gradle.api.tasks.Input}, {@link org.gradle.api.tasks.InputDirectory}, {@link org.gradle.api.tasks.InputFile}, {@link org.gradle.api.tasks.InputFiles}, {@link org.gradle.api.tasks.Internal} or {@link org.gradle.api.model.ReplacedBy}.
```

Comment:
```suggestion
 *     All getters must be annotated with one of the input annotations {@link org.gradle.api.tasks.Console}, {@link org.gradle.api.tasks.Input}, {@link org.gradle.api.tasks.InputDirectory}, {@link org.gradle.api.tasks.InputFile}, {@link org.gradle.api.tasks.InputFiles}, {@link org.gradle.api.tasks.Internal} or {@link org.gradle.api.model.ReplacedBy}.
```

How are we going to not forget adding any new annotation here?

Reply:
I changed the phrase to reference the userguide table.